IPython/kernel/comm/comm.py

```diff
@@ -63,6 +66,10 @@ def __init__(self, target_name='', data=None, **kwargs):
     
     def _publish_msg(self, msg_type, data=None, metadata=None, buffers=None, **keys):
         """Helper for sending a comm message on IOPub"""
+        if threading.current_thread().name != 'MainThread' and IOLoop.initialized():
+            # make sure we never send on a zmq socket outside the main IOLoop thread
+            IOLoop.instance().add_callback(lambda : self._publish_msg(msg_type, data, metadata, buffers, **keys))
```

Comment:
Overflow was a poor choice of words, sorry.  I meant expand until >100% memory usage, but it sounds like that isn't a problem.